### PR TITLE
Changed URLs to HTTPS

### PR DESCRIPTION
**Pull request by einsteinsfool**
Ran `grep -rnw . --exclude-dir={external,build} -e 'http:/'` in the repo directory and changed all occurrences of "http://" to "https://" if the page was accessible via HTTPS.

Sometimes I changed the URL entirely if it was outdated. This was the case for Qt docs, `monerochain.info` in `tree-hash.c` and `GoogleTestAdvancedGuide` in `gtest.cc`.

Let's encrypt the web!

### CODE REVIEW

**Comment by jtgrassie** [comment]
Have all these URL changes been checked?

**Comment by einsteinsfool** [comment]
@moneromooo-monero As for gtest, you can see that the latest commit fixed spelling mistakes in comments. If that's on the table then why not also fix the URLs? Same for epee, some typos were fixed. As for ref10 and in fact gtest and epee too, what bad could happen from changing the URLs in comments? They are only comments and even if there's some new version that overrides the URLs - so be it. It's better to have HTTPS at least for some time. Am I not seeing here some problem?

@jtgrassie Yes, I checked all URLs. There have been some URLs with HTTP that didn't work with HTTPS and I didn't change them. You can check which by pulling my PR and running the grep command from the 1st post. Also, some HTTPS URLs don't work but neither did their HTTP equivalent. They are just outdated and I wasn't able to find a live version for them (as for Qt Docs and GoogleTestAdvancedGuide). I still plan to check all URLs in the repo and find some live versions for dead link, but that's after this PR. And I changed dead URLs to HTTPS too, because why not?

**Comment by moneromooo-monero** [comment]
Can the changes to the RFC be removed ? Also the licence *unless* there's a new version of the licence with those changes.

**Comment by moneromooo-monero** [comment]
ping - need to restore "canonical" files please.

**Comment by einsteinsfool** [comment]
I think I did it. If something's wrong I'll remove my repo and make the PR from scratch with your suggestions as I'm not master of git.

Removed RFC, gtest, epee and licence changes.

Sorry it taked so long!

File: tests/data/fuzz/parse-url/URL2 [hunk 1]
```diff
@@ -1 +1 @@
-iframe_test.html?api_url=http://api.vk.com/api.php&api_id=3289090&api_settings=1&viewer_id=562964060&viewer_type=0&sid=0aad8d1c5713130f9ca0076f2b7b47e532877424961367d81e7fa92455f069be7e21bc3193cbd0be11895&secret=368ebbc0ef&access_token=668bc03f43981d883f73876ffff4aa8564254b359cc745dfa1b3cde7bdab2e94105d8f6d8250717569c0a7&user_id=0&group_id=0&is_app_user=1&auth_key=d2f7a895ca5ff3fdb2a2a8ae23fe679a&language=0&parent_language=0&ad_info=ElsdCQBaQlxiAQRdFUVUXiN2AVBzBx5pU1BXIgZUJlIEAWcgAUoLQg==&referrer=unknown&lc_name=9834b6a3&hash=
\ No newline at end of file
+iframe_test.html?api_url=https://api.vk.com/api.php&api_id=3289090&api_settings=1&viewer_id=562964060&viewer_type=0&sid=0aad8d1c5713130f9ca0076f2b7b47e532877424961367d81e7fa92455f069be7e21bc3193cbd0be11895&secret=368ebbc0ef&access_token=668bc03f43981d883f73876ffff4aa8564254b359cc745dfa1b3cde7bdab2e94105d8f6d8250717569c0a7&user_id=0&group_id=0&is_app_user=1&auth_key=d2f7a895ca5ff3fdb2a2a8ae23fe679a&language=0&parent_language=0&ad_info=ElsdCQBaQlxiAQRdFUVUXiN2AVBzBx5pU1BXIgZUJlIEAWcgAUoLQg==&referrer=unknown&lc_name=9834b6a3&hash=
```
Comment (moneromooo-monero) [inline tests/data/fuzz/parse-url/URL2]: That adds a newline, which is not a valid character in a url. I think (not 100% sure) it's a bad idea to add it here,

**Comment by moneromooo-monero** [comment]
The RFC changes are still here (some of them anyway). Also please squash the changes when done.

**Comment by einsteinsfool** [comment]
What do you mean by squashing the changes? That I should revert the last commit, make changes to RFC and remove that endline and then make a commit instead of adding a 3rd one?

**Comment by moneromooo-monero** [comment]
I mean merging the two commits, where one does things, and the other reverts part of it.
git rebase -i HEAD~2
replace the second "pick" with "f"
save and exit

**Comment by einsteinsfool** [comment]
I'll try.
As for RFC, I did ctrl+f and there was only 1 URL change for RFC - http://rfc.zeromq.org/spec:22. What changes did I miss?

**Comment by moneromooo-monero** [comment]
Click on "Files changed" near the top, you can see a few changes in ietf.txt.

**Comment by einsteinsfool** [comment]
Too difficult with this squashing. Couldn't do it.
Maked a new PR #4045. Sorry for the obstruction.

**Pull request closed without merging**